frontend/src/metabase/containers/Form.jsx

```diff
@@ -243,6 +247,8 @@ class Form extends React.Component {
         };
       }
     } finally {
+      const fieldNames = this._getFieldNames();
+      untouch(...fieldNames);
```

Comment:
I tested that we don't really use `touched` of the form. So, it should be okay. But I'd want to hear other engineers' opinions on this. @kulyk @ariya @ranquild .

Otherwise, I could just only untouch all fields when `shouldPersistError: true`.

Reply:
Guess I haven't seen `touched` used anywhere. The thing is we use `pristine` property which is just the opposite of `touched`, so this potentially can affect those